types/wlr_screenshooter.c

```diff
@@ -127,17 +120,12 @@ static void screenshooter_shoot(struct wl_client *client,
 	if (!state) {
 		wl_resource_destroy(screenshot->resource);
 		free(screenshot);
-		free(pixels);
 		wl_resource_post_no_memory(screenshooter_resource);
```

Comment:
Doesn't posting no memory to a client cause it to be disconnected? In that case you don't need to destroy the resource above if it belongs to that client because it will get cleaned up anyway (I may be wrong).

Reply:
Meh, the code is more readable that way.